test: add e2e tests for static examples

lag-of-death opened the pull request:
Add tests for static examples - closes #414

XVincentX commented on test-harness/specs/static_examples-AC-1.oas3.txt:
```diff
@@ -0,0 +1,52 @@
+====spec====
+info:
+  title: Sample API
+  version: 1.0.0
```
You've been skipping this in the other documents, right? Maybe we can remove it from here too?

XVincentX commented on test-harness/specs/static_examples-AC-1.oas3.txt:
```diff
@@ -0,0 +1,52 @@
+paths:
+  /todos:
+    get:
+      parameters:
```
I think the parameter here is not required, we can remove it.

XVincentX commented on test-harness/specs/static_examples-AC-2.oas3.txt:
```diff
@@ -0,0 +1,52 @@
+  /todos:
+    get:
+      parameters:
+        - name: title
```
Dtto, can we remove this part?

XVincentX commented on test-harness/specs/static_examples-AC-2.oas3.txt:
```diff
@@ -0,0 +1,52 @@
+====spec====
+info:
+  title: Sample API
+  version: 1.0.0
```
Yeah I think we can remove this

philsturgeon commented on test-harness/specs/static_examples-AC-1.oas2.txt:
```diff
@@ -0,0 +1,42 @@
+====test====
+Given
```
I don't understand why we would do this? This seems to be duplicating information from the rest of the file, and I am not sure what benefit it serves? 

Yesterday we talked about simplifying the `==test==` content by making it more human, less gherkin, and avoiding duplication with the test file. This seems to have made it less human, more gherkin, increased duplication, and added some blank lines in for unknown reasons. 

> When prism is started with the --dynamic flag, responses should continue to match expected document structure

Done :)

lag-of-death commented:
@XVincentX, I cleaned up the specs and also fixed some false positives. Turns out, that when there was no `-i` in `curl something`, then, the bodies from https://github.com/stoplightio/prism/blob/21ba780a85ef20a9313286dd511af18965ff591d/test-harness/index.ts#L62 were empty

XVincentX commented:
@lag-of-death Ok that's the part I should check out in the harness, apparently when no -i the body is…gone. In meantime I've added the flag to all the missing tests so we tests the things :)
